src/Adafruit_SPIFlashBase.cpp

```diff
@@ -99,7 +99,7 @@ static const SPIFlash_Device_t possible_devices[] = {
     MB85RS64V, MB85RS1MT, MB85RS2MTA, MB85RS4MT,
 
     // Other common flash devices
-    W25Q16JV_IQ, W25Q32JV_IQ, AT25SF041, AT25DF081A};
+    W25Q16JV_IQ, W25Q32JV_IQ, AT25SF041, AT25DF081A, IS25LP016D};
```

Comment:
please remove this from default/possible devices since we cannot support all devices (consume ROM). You could pass it as argument to begin().

Reply:
humm... about that I have a proposal. In order to mantain a native support of 3d party boards' flash memories without needing using the custom `begin()` call function: on `variant.h` files for each board already is usually defined a macro named `EXTERNAL_FLASH_DEVICES`.

So in `variant.h `files we could have for example this line defined:

`#define EXTERNAL_FLASH_DEVICES  IS25LP016D, GD25Q16C`

And later in `Adafruit_SPIFlashBase.cpp` file we could use that macro by do the following:
```
#if defined (EXTERNAL_FLASH_DEVICES)
/// List of all possible flash devices defined by variant.h
static const SPIFlash_Device_t possible_devices[] = {EXTERNAL_FLASH_DEVICES};
#else
/// List of all possible flash devices used by Adafruit boards
static const SPIFlash_Device_t possible_devices[] = {
    // Main devices used in current Adafruit products
    GD25Q16C, GD25Q32C, GD25Q64C, S25FL116K, S25FL216K,
....
#endif
```
With this change we can even reduce more the ROM consumtion.
What do you think? :)

Reply:
The backup blan would be to also search the jedec id on `EXTERNAL_FLASH_DEVICES` if defined:

```
bool Adafruit_SPIFlashBase::begin(SPIFlash_Device_t const *flash_devs,
                                  size_t count) {
...
  // Check for device in variant.h supplied list, if any.
  #ifdef EXTERNAL_FLASH_DEVICES
  if (_flash_dev == NULL) {
    SPIFlash_Device_t external_possible_devices[] = {EXTERNAL_FLASH_DEVICES};
    int num_external_flash_devices = sizeof(external_possible_devices) / sizeof(external_possible_devices[0]);
    _flash_dev = findDevice(external_possible_devices, num_external_flash_devices, jedec_ids);
  }
  #endif
...
}
```

Reply:
@nekuneko the first suggestion is better, please make the changes. Though since it could potentially break some of existing codes, it will take me a bit of time to do an overall review of all variants and libs using this.

Reply:
Done :). Thank you very much for the clang-format tip! There is no hurry take it easy to review, I understand these changes could break some code, if so I'm sure we can fix it.